hack/convert.go

```diff
@@ -0,0 +1,180 @@
+		for _, resource := range group.Resources {
+			resources = append(resources, resource.Kind)
+		}
+	}
+	moduleRoot := util.GetModuleRoot()
+	err := filepath.Walk(
+		filepath.Join(moduleRoot, "api/gloo"),
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+			if !strings.HasSuffix(info.Name(), ".proto") {
+				return nil
+			}
+			file, err := os.OpenFile(path, os.O_RDWR, os.ModePerm)
+			if err != nil {
+				return err
+			}
+			defer file.Close()
+			wholeFileBytes, err := ioutil.ReadAll(file)
+			if err != nil {
+				return err
+			}
+			// If the proto file is not from a relevant package skip it.
+			wholeFileBytes, ok := isRelevantFile(wholeFileBytes)
+			if !ok {
+				return nil
+			}
+
+			var fileByLine []string
+			scan := bufio.NewScanner(bytes.NewBuffer(wholeFileBytes))
+			for scan.Scan() {
+				line := scan.Text()
+				if strings.Contains(line, "// Metadata contains the object metadata for this resource") ||
+					strings.Contains(line, "// Status is read-only by clients, and set by gloo during validation") ||
+					strings.Contains(line, "// Status indicates the validation status of this resource.") ||
+					strings.Contains(line, "option (core.solo.io.resource)") ||
+					strings.Contains(line, "solo-kit/api/v1/status.proto") ||
+					strings.Contains(line, "solo-kit/api/v1/metadata.proto") ||
+					strings.Contains(line, "core.solo.io.Metadata") ||
+					strings.Contains(line, "core.solo.io.Status") {
+					continue
+				}
+
+				fileByLine = append(fileByLine, line)
+			}
+			err = file.Truncate(0)
+			if err != nil {
+				return err
+			}
+			_, err = file.Seek(0, 0)
+			if err != nil {
+				return err
+			}
+			_, err = file.WriteString(strings.Join(fileByLine, "\n"))
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		log.Printf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func isRelevantFile(file []byte) ([]byte, bool) {
+	var relevantMessage bool
+	glooGroups := codegen.GlooGroups()
+	relevantTypes := getRelevantTypes(file, glooGroups)
+	// If there are no relevant types in this file, return
+	if len(relevantTypes) == 0 {
+		return nil, false
+	}
+	for _, relevantType := range relevantTypes {
+		// Make sure to add extra space after to check specifically for only Message
+		oldMessageBytes := []byte(fmt.Sprintf("message %s ", relevantType))
+		if bytes.Contains(file, oldMessageBytes) {
+			oldMessageBytes = oldMessageBytes[:len(oldMessageBytes)-1]
+			file = bytes.ReplaceAll(file, oldMessageBytes, append(oldMessageBytes, []byte("Spec")...))
+			statusBytes := &bytes.Buffer{}
+			if err := statusTemplate.Execute(statusBytes, relevantType); err != nil {
+				panic(err)
+			}
+			file = append(file, statusBytes.Bytes()...)
+			file = addImport(file)
+			relevantMessage = true
+		}
+	}
+
+	return file, relevantMessage
+}
+
+func getRelevantTypes(file []byte, glooGroups []model.Group) []string {
```

Comment:
Can we add a comment that explains what the returned list of strings represents (the kind names of all gloo resources)?